kibana-reports/server/model/index.ts

```diff
@@ -61,62 +64,100 @@ export const cronSchema = schema.object({
+  report_params: schema.object({
+    report_name: schema.string(),
+    report_source: schema.oneOf([
+      schema.literal(REPORT_TYPE.dashboard),
+      schema.literal(REPORT_TYPE.visualization),
+      schema.literal(REPORT_TYPE.savedSearch),
+    ]),
+    description: schema.string(),
+
+    core_params: schema.conditional(
+      schema.siblingRef('report_source'),
+      REPORT_TYPE.savedSearch,
+      dataReportSchema,
+      visualReportSchema
+    ),
+  }),
 
   delivery: schema.maybe(
     schema.object({
-      channel: schema.oneOf([
-        schema.literal('Email'),
-        schema.literal('Slack'),
-        schema.literal('Chime'),
-        schema.literal('Kibana User'),
+      recipients: schema.arrayOf(schema.string(), { minSize: 0 }),
+      title: schema.string(),
+      description: schema.oneOf([
+        schema.object({ text: schema.string() }),
+        schema.object({ html: schema.string() }),
       ]),
-      //TODO: no validation on delivery settings for now, because @kbn/config-schema has no support for more than 2 conditions
-      delivery_params: schema.any(),
+      channel_ids: schema.maybe(schema.arrayOf(schema.string())),
     })
   ),
 
-  trigger: schema.maybe(
-    schema.object({
-      trigger_type: schema.oneOf([
-        schema.literal('Alert'),
-        schema.literal('Schedule'),
-        schema.literal('On demand'),
-      ]),
-      trigger_params: schema.conditional(
-        schema.siblingRef('trigger_type'),
-        'Alert',
-        // TODO: add alerting schema here once we finished the design for alerting integration
-        schema.any(),
-        scheduleSchema
-      ),
-    })
+  trigger: schema.object({
+    trigger_type: schema.oneOf([
+      /*
+        TODO: Alerting will be added in the future.
+        Currently @kbn/config-schema has no support for more than 2 conditions, keep an eye on library update
+      */
+      // schema.literal(TRIGGER_TYPE.alerting)
+      schema.literal(TRIGGER_TYPE.schedule),
+      schema.literal(TRIGGER_TYPE.onDemand),
+    ]),
+    trigger_params: schema.conditional(
+      schema.siblingRef('trigger_type'),
+      TRIGGER_TYPE.onDemand,
+      schema.never(),
+      scheduleSchema
+    ),
+  }),
+
+  time_created: schema.maybe(schema.number()),
+  last_updated: schema.maybe(schema.number()),
+  status: schema.maybe(
+    schema.oneOf([
+      schema.literal(REPORT_DEFINITION_STATUS.active),
+      schema.literal(REPORT_DEFINITION_STATUS.disabled),
+    ])
+  ),
+});
+
+export const reportSchema = schema.object({
+  query_url: schema.uri(),
+  time_from: schema.number(),
+  time_to: schema.number(),
+  report_definition: reportDefinitionSchema,
+
+  time_created: schema.maybe(schema.number()),
+  state: schema.maybe(
+    schema.oneOf([
+      schema.literal(REPORT_STATE.created),
+      schema.literal(REPORT_STATE.error),
+    ])
   ),
 });
 
+export type ReportDefinitionSchemaType = TypeOf<typeof reportDefinitionSchema>;
```

Comment:
Is there a reason we are not consistently using camelCase with these type exports?

Reply:
if you mean the fields in the schema are in snake case, that's following the ES data fields convention